### PR TITLE
docs(config): add `experimentalScopedSlotChanges` extras flag

### PR DESCRIPTION
**Pull request by tanner-reits**
Adds a new `experimentalScopedSlotChanges` extras config option to toggle the new behavior for `scoped` components.

Related code changes: https://github.com/ionic-team/stencil/pull/5220

**Note:** The first commit only alphabetizes the options, but results in what looks like many more changes

### CODE REVIEW

File: docs/config/extras.md [hunk 1]
```diff
@@ -63,30 +63,23 @@ Users of this flag should note that they may see an increase in their bundle siz
...
-| `stencil_componentWillRender` | Dispatched for each component's `componentWillRender`. |
-| `stencil_componentDidLoad`    | Dispatched for each component's `componentDidLoad`.    |
-| `stencil_componentDidUpdate`  | Dispatched for each component's `componentDidUpdate`.  |
-| `stencil_componentDidRender`  | Dispatched for each component's `componentDidRender`.  |
+When set to `true`, the following behaviors will be applied:
 
-### scopedSlotTextContentFix
-
-An experimental flag that when set to `true`, aligns the behavior of invoking the `textContent` getter/setter on a scoped component to act more like a component that uses the shadow DOM. Specifically, invoking `textContent` on a component will adhere to the return values described in [MDN's article on textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#description). Defaults to `false`.
+<!-- TODO(NOW): confirm fix versions -->
```
Comment (tanner-reits) [inline docs/config/extras.md:73]: I'll make sure these versions are accurate before this merges, just depends how this lines up with upcoming releases

**Review by rwaskiewicz — COMMENTED** [review]
Overall LGTM - I'll give it another pass once I get to/through the impl PRs

**Review by christian-bromann — APPROVED** [review]
👍

**Review by rwaskiewicz — APPROVED** [review]
I approved the impl PR, I'm good with approving now and just landing it after you've bumped the version in the docs

**Pull request merged**